[DOCS-1683] password reset behavior (#9744)
DataDog/documentation: content/en/account_management/faq/password-requirements.md

1. Passwords have to be a minimum of 8 characters and a maximum of 70 characters.
2. Passwords must contain at least 1 lowercase character and 1 number.

> 3. If you reset your password, you cannot use the same password twice.

Reviewer: ```suggestion
3. You cannot reset your password to one you've used in the past.
```
